opcua/client/client.py

```diff
@@ -256,6 +267,10 @@ def connect(self):
         self.connect_socket()
         try:
             self.send_hello()
+        except ua.UaStatusCodeError:
+            raise
+
+        try:
```

Comment:
by moving send_hello in its own try/Catch you are not closing correctly the socket

Reply:
and catching an exception for then re-raising it immediatly without doing anything, does not make sense

Reply:
Good point. I've removed the unnecessary try-except and now see the socket being closed in the logs. Thank you!

```
cryptography is not installed, use of crypto disabled
cryptography is not installed, use of crypto disabled
cryptography is not installed, use of crypto disabled
2018-12-05 14:17:21.634 [opcua.client.ua_client.Socket] 140735838761856 INFO - opening connection
2018-12-05 14:17:21.637 [opcua.client.ua_client.Socket] 123145482985472 INFO - Thread started
2018-12-05 14:17:21.638 [opcua.uaprotocol] 123145482985472 DEBUG - Waiting for header
2018-12-05 14:17:21.646 [opcua.uaprotocol] 123145482985472 INFO - received header: Header(type:b'ERR', chunk_type:b'F', body_size:123, channel:0)
2018-12-05 14:17:21.646 [opcua.uaprotocol] 123145482985472 DEBUG - Waiting for header
2018-12-05 14:17:21.646 [opcua.client.ua_client.Socket] 140735838761856 INFO - Request to close socket received
2018-12-05 14:17:21.647 [opcua.client.ua_client.Socket] 123145482985472 INFO - Socket has closed connection
2018-12-05 14:17:21.647 [opcua.client.ua_client.Socket] 123145482985472 INFO - Thread ended
2018-12-05 14:17:21.647 [opcua.client.ua_client.Socket] 140735838761856 INFO - Socket closed, waiting for receiver thread to terminate...
2018-12-05 14:17:21.647 [opcua.client.ua_client.Socket] 140735838761856 INFO - Done closing socket: Receiving thread terminated, socket disconnected
2018-12-05 14:17:21.647 [__main__] 140735838761856 INFO - OPC UA server responded with error BadTcpEndpointUrlInvalid: The server does not recognize the QueryString specified.
```